packages/components/image/src/image.vue

```diff
@@ -61,6 +67,17 @@ export default defineComponent({
   emits: imageEmits,
 
   setup(props, { emit, attrs: rawAttrs }) {
+    useDeprecated(
+      {
+        scope: 'el-image',
+        from: 'append-to-body',
+        replacement: 'preview-append-to-body',
+        version: '2.2.0',
+        ref: 'https://element-plus.org/en-US/component/image.html#image-attributess',
+      },
+      props.appendToBody
+    )
```

Comment:
After warning about the deprecation, we should use a compatible version for this variable:

```ts
const compatibleAppendToBody = computed(() => {
  isBoolean(props.appendToBody) ? props.appendToBody : props.previewAppendToBody
})

```